src/Blockcore.Indexer.Core/Sync/SyncTasks/BlockPuller.cs

```diff
@@ -73,7 +72,7 @@ public override async Task<bool> OnExecute()
             return true;
          }
 
-         if (Runner.GlobalState.Blocked)
+         if (Runner.GlobalState.Blocked || Abort)
```

Comment:
no need to put Abort here becuase the underline runner checks that (I will fix it)

Reply:
ah actually this is correct it will abort the loop